pyro/ops/gaussian.py

```diff
@@ -207,6 +207,49 @@ def event_logsumexp(self):
                 chol_P.diagonal(dim1=-2, dim2=-1).log().sum(-1))
 
 
+class AffineNormal:
+    """
+    Represents a conditional diagonal normal distribution over a random
+    variable ``Y`` whose mean is an affine function of a random variable ``X``.
+    The likelihood of ``X`` is thus::
+
+        LinearNormal(matrix, loc, scale).condition(y).log_density(x)
```

Comment:
nit: LinearNormal -> AffineNormal
You might want to use the name `MultivariateNormalDiag` as in tensorflow (though I don't like this idea much).

Reply:
Done.

I agree that `AffineNormal` is a weird name, but I don't think `MultivariateNormalDiag` is the right name either. This class really serves a very narrow role as a likelihood of a diag normal observation of an affine transformation. I'm open to renaming / refactoring once we see whether and how this pattern generalizes.